cviceni7/petrekova07/tuples_and_sets.ipynb

```diff
@@ -0,0 +1,1017 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 18,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "---\n",
+      "Novy radek s delkou 3, max hodnota je 3\n",
+      "1\n",
+      "2\n",
+      "3\n",
+      "---\n",
+      "---\n",
+      "Novy radek s delkou 3, max hodnota je 6\n",
+      "4\n",
+      "5\n",
+      "6\n",
+      "---\n",
+      "---\n",
+      "Novy radek s delkou 3, max hodnota je 9\n",
+      "7\n",
+      "8\n",
+      "9\n",
+      "---\n"
+     ]
+    }
+   ],
+   "source": [
+    "# A používat vnořené smyčky a vestavěné funkce, které známe z listů\n",
+    "for radek in vnořený_tuple:\n",
+    "    print(\"---\")\n",
+    "    print(f\"Novy radek s delkou {len(radek)}, max hodnota je {max(radek)}\")\n",
+    "    for prvek in radek:\n",
+    "        print(prvek)\n",
+    "    print(\"---\")\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 19,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Radek cislo 0 s hodnotami (1, 2, 3)\n",
+      "Radek cislo 1 s hodnotami (4, 5, 6)\n",
+      "Radek cislo 2 s hodnotami (7, 8, 9)\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Či použijeme for i cyklus\n",
+    "for i in range(len(vnořený_tuple)):\n",
+    "    print(f\"Radek cislo {i} s hodnotami {vnořený_tuple[i]}\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 20,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "529344067295497451\n",
+      "529344067295497451\n",
+      "-4363729961677198915\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Nezapomínejme, že tuple s neměnnými prvky je hashovatelný\n",
+    "tuple_A = (1, 2, 3)\n",
+    "tuple_B = (1, 2, 3)\n",
+    "tuple_C = (1, 2, 4)\n",
+    "\n",
+    "print(hash(tuple_A))\n",
+    "print(hash(tuple_B))\n",
+    "print(hash(tuple_C))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 21,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Tuple_A a Tuple_B jsou stejné\n"
+     ]
+    }
+   ],
+   "source": [
+    "# To můžeme využít k porovnání dvou tuple\n",
+    "if hash(tuple_A) == hash(tuple_B):\n",
+    "    print(\"Tuple_A a Tuple_B jsou stejné\")\n",
+    "\n",
+    "if hash(tuple_A) == hash(tuple_C):\n",
+    "    print(\"Tuple_A a Tuple_C jsou stejné\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 23,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "(1, 2, [3, 4])\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Pozor, pokud tuple obsahuje mutable prvky, tak už není hashovatelný\n",
+    "tuple_D = (1, 2, [3, 4])\n",
+    "\n",
+    "#print(hash(tuple_D)) # chyba\n",
+    "print(tuple_D)"
```

Comment:
Líbí se mi, jak sis tu chybu zakomentovala, aby ti to fungovalo, na to většina kašlala:D